packages/lwc/scripts/utils/rollup.js

```diff
@@ -66,7 +66,7 @@ function rollupConfig(config) {
             name,
             file: path.join(targetDirectory, target, generateTargetName(config)),
             format,
-            plugins: [prod && !debug && rollupTerser()],
+            plugins: [prod && !debug && minify()],
```

Comment:
The build process is quite heavy today. We spin up multiple workers and in each worker, we compile each package in parallel using rollup. I am wondering if we can completely migrate the build step to esbuild. As far as I understand esbuild API, it can do both es5 and the `process.env.NODE_ENV` replacement. The only blocker might be the LWC feature transform.

Reply:
I actually have a branch where I am experimenting with replacing typescript with esbuild. It's 0m21.100s as opposed to 0m27.716s on master (**6.6s** faster!).

The problem is that `esbuild` does not generate `.d.ts` files – it just strips types. This fails the build since e.g. `@lwc/shared` has to generate `.d.ts` files for other internal packages to consume. There doesn't seem to be a good fix for this: https://github.com/egoist/rollup-plugin-esbuild#declaration-file

Plus, we would still have to run TypeScript in CI so that it can do typechecking. ESBuild does not do typechecking.

However, I think we could use `esbuild` _just_ for the `lwc` package. It doesn't need to generate declaration files or do typechecking.

Reply:
OK, now that the LWC features transform is out of `lwc`, then yes, we can replace all of Rollup+Babel+Terser in `lwc` with esbuild. Or swc. Or whatever. That could save us ~10s on the build time. Maybe we should focus on doing that.

Reply:
I just looked into it. ESBuild doesn't support all ES5 transforms we need (const -> var, destructuring) (https://github.com/evanw/esbuild/issues/988) and also doesn't support UMD format (https://github.com/evanw/esbuild/issues/507).

I think we might as well just cover this when we do #2430 and remove all the non-ESM stuff, and when we can retire IE support. Closing for now.

Reply:
If we _do_ generate only ES2017 files (https://github.com/nolanlawson/lwc-1/commit/c23fa19c5281337eb8a92921d13d5ea3648657bf), then ESBuild can do it in 0.162s. This is the shiny future that awaits us! ✨